src/fosslight_util/download.py

```
@@ -1013,6 +1013,74 @@ def extract_rpm_payload(source_file: str, dest_path: str) -> bool:
+        try:
+            os.chmod(p, os.stat(p).st_mode | 0o644)
+        except Exception:
+            pass
+
+
+def _tar_extractall_safe(fname: str, open_mode: str, extract_path: str) -> None:
+    members = None
+    try:
+        with contextlib.closing(tarfile.open(fname, open_mode)) as tf:
+            members = tf.getmembers()
+            tf.extractall(path=extract_path)
```

Ruff (0.15.15)

[error] Uses of `tarfile.extractall()`

(S202)